### DISCUSSION TITLE
jump to implicit argument definition

### DISCUSSION THREAD

**Discussion by antosha417** [scalameta/nvim-metals#274] [Q&A] [answered]
In intellij idea there is an action called 'show implicit arguments'. If I click on element in implict arguments list I jump to the location where this argument is defined. Is there a way to tell where implict arguments come from in nvim-metals?

**Comment by ckipp01** [comment, marked as answer]
Hey @antosha417, great question. So there is a way to _see_ which are being used. You can see this if you set `showImplicitArguments = true`. Then when you do a hover, it will be in the `With synthetics added` section. Here is a simple example:

<img width="445" alt="Screenshot 2021-11-17 at 12 21 11" src="https://user-images.githubusercontent.com/13974112/142191337-e9047a37-4171-4383-a072-bf9095ae8b80.png">

While this does show you where your implicit is coming from, it doesn't really allow you to _jump_ to it. The easiest thing to do would probably be the "goto workspace symbol", and then you can just type the symbol in, and you should be able to jump to it that way.

There are actually two settings that might interest you related to implicits:

https://github.com/scalameta/nvim-metals/blob/152af730d398e560c9722d6e0de19cf3d2ff3887/doc/metals.txt#L313-L324

**Reply by antosha417** [reply]
Thank you for a fast reply!

In my case in hover I have `clientLockService: ClientLockService[F]`
and with synthetics added I have `clientLockService$implicit$1` 
I guess it means that this `clientLockService` is defined in the same for-yeld term. 

Another cases are working as you described, thanks again!
